bob-backend/src/pearl/group.rs

```diff
@@ -261,6 +265,7 @@ impl Group {
             .iter_possible_childs_rev(&Key::from(key))
             .map(|(_, x)| &x.data)
         {
+            let _reinit_lock = self.reinit_lock.try_read().map_err(|_| Error::holder_temporary_unavailable())?;
```

Comment:
This should be at the beginning of the function (you should protect `holders` iteration as well)